src/services/smart-fields-values-injector.js

```diff
@@ -33,6 +33,19 @@ function SmartFieldsValuesInjector(record, modelName, fieldsPerModel) {
         if (_.isFunction(value.then)) {
           return value
             .then((result) => {
+              // NOTICE: We need some recursion in order to consider smart fields in smart
```

Comment:
[Food for thought] Recursivity is powerful but could also increase the response time of the request.
Actually whatever the way it is fixed it will increase the response time.

For instance, if you have a Smart Relationship with asynchronous requests, and a pagination of 20 records:
- with the current code base you will have 20 asynchronous calls
- with this code, you'll have 20 asynchronous calls x number of Smart Fields declared in the referenced model.

Reply:
Yes, sadly 😢

```
number_of_records_paginated * number_of_smart_fields_that_returns_a_collection
```

I don't know how to avoid this if we want to consider smart fields in smart relations. We have to keep this caveat in mind though.